src/widget/woverview.cpp

```diff
@@ -313,28 +381,119 @@ void WOverview::receiveCuesUpdated() {
 }
 
 void WOverview::mouseMoveEvent(QMouseEvent* e) {
-    if (m_orientation == Qt::Horizontal) {
-        m_iPos = math_clamp(e->x(), 0, width() - 1);
-    } else {
-        m_iPos = math_clamp(e->y(), 0, height() - 1);
+    // Do not activate cue hovering while right click is held down and the
+    // button down event was not on a cue.
+    if (m_bTimeRulerActive) {
+        // Prevent showing times beyond the boundaries of the track when the
+        // cursor is dragged outside this widget before releasing right click.
+        m_timeRulerPos.setX(math_clamp(e->pos().x(), 0, width()));
+        m_timeRulerPos.setY(math_clamp(e->pos().y(), 0, height()));
+        update();
+        return;
+    }
+
+    m_pHoveredMark.clear();
+    // Without some padding, the user would only have a single pixel width that
+    // would count as hovering over the WaveformMark.
+    float lineHoverPadding = 5.0;
+    // Non-hotcue marks (intro/outro cues, main cue, loop in/out) are sorted
+    // before hotcues in m_marksToRender so if there is a hotcue in the same
+    // location, the hotcue gets rendered on top. When right clicking, the
+    // the hotcue rendered on top must be assigned to m_pHoveredMark to show
+    // the CueMenu. To accomplish this, m_marksToRender is iterated in reverse
+    // and the loop breaks as soon as m_pHoveredMark is set.
+    for (auto it = m_marksToRender.crbegin(); it != m_marksToRender.crend(); ++it) {
```

Comment:
Building fails here crbegin() is new in Qt 5.6. I am afraid you need to switch to an index based loop.

Reply:
```
    for (int i = m_marksToRender.size(); i >= 0 ; --i) {
        WaveformMarkPointer pMark = m_marksToRender.at(i);
```